From 4c0336d2f02fc144e0f7902094f1198220412a6a Mon Sep 17 00:00:00 2001
From: Wang Xuerui <idontknw.wang@gmail.com>
Date: Sun, 20 Sep 2015 23:49:49 +0800
Subject: [PATCH] gps: use MTK gps header

... thanks to @DerTeufel!

[Ported to N by @xen0n. Major changes required in HAL code.]

Change-Id: I9695e05edc0a39f3593ac5775b8e98b005339f9f
---
 include/hardware/gps.h | 40 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/include/hardware/gps.h b/include/hardware/gps.h
index acf601b..b0b725e 100644
--- a/include/hardware/gps.h
+++ b/include/hardware/gps.h
@@ -37,6 +37,9 @@ __BEGIN_DECLS
 /** Milliseconds since January 1, 1970 */
 typedef int64_t GpsUtcTime;
 
+/** Maximum number of MTK legacy GNSS SVs for gps_sv_status_callback(). */
+#define MTK_LEGACY_GNSS_MAX_SVS 256
+
 /** Maximum number of SVs for gps_sv_status_callback(). */
 #define GPS_MAX_SVS 32
 /** Maximum number of SVs for gps_sv_status_callback(). */
@@ -572,6 +575,32 @@ typedef struct {
     float   azimuth;
 } GpsSvInfo;
 
+/** Represents MTK legacy GNSS SV information. */
+typedef struct {
+    /** set to sizeof(MTKLegacyGnssSvInfo) */
+    size_t          size;
+    /** Pseudo-random number for the SV. */
+    int     prn;
+    /** Signal to noise ratio. */
+    float   snr;
+    /** Elevation of SV in degrees. */
+    float   elevation;
+    /** Azimuth of SV in degrees. */
+    float   azimuth;
+    /**
+     * SVs have ephemeris data.
+     */
+    bool has_ephemeris;
+    /**
+     * SVs have almanac data.
+     */
+    bool has_almanac;
+    /**
+     * SVs were used for computing the most recent position fix.
+     */
+    bool used_in_fix;
+} MTKLegacyGnssSvInfo;
+
 typedef struct {
     /** set to sizeof(GnssSvInfo) */
     size_t size;
@@ -652,6 +681,17 @@ typedef struct {
 
 } GnssSvStatus;
 
+typedef struct {
+    /** set to sizeof(MTKLegacyGnssSvStatus) */
+    size_t          size;
+
+    /** Number of SVs currently visible. */
+    int         num_svs;
+
+    /** Contains an array of GNSS SV information. */
+    MTKLegacyGnssSvInfo   sv_list[MTK_LEGACY_GNSS_MAX_SVS];
+} MTKLegacyGnssSvStatus;
+
 /* CellID for 2G, 3G and LTE, used in AGPS. */
 typedef struct {
     AGpsRefLocationType type;
-- 
2.11.0

